netbox/netbox/tables/columns.py

```diff
@@ -51,7 +52,9 @@ class DateColumn(tables.DateColumn):
     default, making this behavior consistent in all fields of type DateField.
     """
     def value(self, value):
-        return value
+        if value:
+            return date_format(value, format="SHORT_DATE_FORMAT")
+        return None
```

Comment:
For `value()`, we want to return the raw ISO-standard date regardless of how the UI display is configured. Instead, this should be applied to the column's `render()` method.